container_pipeline/workers/build.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import Build, get_job_name
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+
+
+class BuildWorker(BaseWorker):
+    """Build worker"""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        super(BuildWorker, self).__init__(logger, sub, pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def handle_job(self, job):
+        """Handle job"""
+        parent_build_running = False
+        parents = job.get('depends_on', '').split(',')
+        parents_in_build = []
+        for parent in parents:
+            is_build_running = Build(parent).is_running()
+            if is_build_running:
+                parents_in_build.append(parent)
+            parent_build_running = parent_build_running or \
+                is_build_running
+
+        if parent_build_running:
+            self.logger.info('Parents in build: %s, pushing job: %s back '
+                             'to queue' % (parents_in_build, job))
+            self.queue.put(json.dumps(job), 'master_tube')
+        else:
+            self.logger.info('Starting build for job: %s' % job)
+            success = self.build(job)
+            if success:
+                self.handle_build_success(job)
+            else:
+                self.handle_build_failure(job)
+
+    def build(self, job):
+        """Build job"""
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        try:
+            self.openshift.login()
+            build_id = self.openshift.build(project, 'build')
+            if not build_id:
+                return False
+        except OpenshiftError as e:
+            logger.error(e)
+            return False
+
+        Build(namespace).start()
```

Comment:
This will be connected to my earlier comment on the line about 
```python
is_build_running = Build(parent).is_running()
```
- Value of `namespace` for the [same index entry](https://github.com/CentOS/container-index/blob/master/index.d/pipeline-images.yml#L2-L11) will look like: `pipeline-images-cccp-build-latest`
- for the object of `utils.Build` class, values of `self.name` and `self._path` will be `pipeline-images-cccp-build-latest` and `/srv/pipeline-logs/pipeline-images-cccp-build-latest` respectively.
- when `start()` method is called, it will create a path `/srv/pipeline-logs/pipeline-images-cccp-build-latest`

Let's see its parent image `centos/centos:latest` from the `depends-on` field. [Here's its index entry](https://github.com/CentOS/container-index/blob/master/index.d/centos.yml#L2-L11).

When this image's build is started, the path it will create upon `Build(namespace).start()` will be: `/srv/pipeline-logs/centos-centos-latest`. But we are checking for `/srv/pipeline-logs/centos/centos:latest` file location in `is_build_running = Build(parent).is_running()`, right?

Unless I'm missing something, this looks a bug to me.